backend/gsr_booking/management/commands/get_reservations.py

```diff
@@ -0,0 +1,90 @@
+import datetime
+
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand
+from django.db.models import Q
+
+from backend.gsr_booking.models import Reservation
+from gsr_booking.models import Group
+
+
+User = get_user_model()
+
+
+class Command(BaseCommand):
+    help = """
+    Get number of reservations using GSR bookings.
+
+    Defaulted to current number. Use --start or --end to specify date range.
+
+    --group     flag to specify group
+    --start     flag to specify start date; expected format: YYYY-MM-DD
+    --end       flag to specify end date; expected format: YYYY-MM-DD
+    --current   flag to specify current reservations
+    --time      flag to specify total time of reservations
+
+    Note: --start/--end and --current are mutually exclusive
+    """
+
+    def add_arguments(self, parser):
+        # optional flags
+        parser.add_argument("--group", type=str, default=None)
+        parser.add_argument("--start", type=datetime, default=None)
+        parser.add_argument("--end", type=str, default=None)
+        parser.add_argument("--current", type=bool, default=False)
+        parser.add_argument("--time", type=bool, default=False)
+
+    def handle(self, *args, **kwargs):
+        group = kwargs["group"]
+        start = kwargs["start"]
+        end = kwargs["end"]
+        current = kwargs["current"]
+        time = kwargs["time"]
+
+        if start and not (start := self.__convert_date(start)):
+            self.stdout.write("Error: invalid start date format")
+            return
+        if end and not (end := self.__convert_date(end)):
+            self.stdout.write("Error: invalid end date format")
+            return
+
+        reservation_filter = Q()
+        if current:
+            reservation_filter &= Q(start__lte=datetime.now()) & Q(end__gte=datetime.now())
+        else:
+            if start:
+                reservation_filter &= Q(start__gte=start)
+            if end:
+                reservation_filter &= Q(end__lte=end)
+
```

Comment:
i don't think current and start/end needs to be mutually exclusive. it might make sense that to look for bookings that started after a certain time but is still going on. the reverse going on now but ended before some time doesn't really make sense to ask... but in any case the logic for both cases can still be achieved (the second would just give 0 bookings) by having default start to be lte now and end gte now, but overrided if start/end are passed, and then having the overall filter to just be the & of these.

Reply:
ok ya that's a fair point